apollo-router/src/axum_http_server_factory.rs

```diff
@@ -2495,4 +2611,64 @@ Content-Type: application/json\r
             assert!(value == "one" || value == "two");
         }
     }
+
+    #[tokio::test]
+    async fn it_makes_sure_same_listenaddrs_are_accepted() {
+        let configuration = Configuration::fake_builder().build();
+
+        init_with_config(MockSupergraphService::new(), configuration, MultiMap::new()).await;
+    }
+
+    #[tokio::test]
+    #[should_panic(
+        expected = "Failed to create server factory: DifferentListenAddrsOnSamePort(127.0.0.1, 0.0.0.0, 4000)"
```

Comment:
what's the reason behind this? Do we consider that a sandbox with a different IP but same port would be a misconfiguration? What about this:
- graphql server listening on `0.0.0.0:4000`
- sandbox listening on `<IP accessible only from the internal network>:4000`

Reply:
We want the test to fail if two different listenaddrs want to bind to the same port. This would be a security issue (binding to 127.0.0.1:4000 and 0.0.0.0:4000 would effectively mean the 127.0.0.1 endpoint is accessible from WAN)

The reason why this test fails this way is because we're using the default graphql endpoint (127.0.0.1:4000) and  0.0.0.0:4000 sandbox, which would "leak the graphql endpoint".

The example you have in mind is currently not supported (this would require more efforts like binding to 0.0.0.0:4000 and then manually "route" the connections according to the configuration, so we decided to not support it yet.

Reply:
> This would be a security issue (binding to 127.0.0.1:4000 and 0.0.0.0:4000 would effectively mean the 127.0.0.1 endpoint is accessible from WAN)

It looks like I don't have the right understanding of this PR then. In which case could this happen? What would it look like in the configuration file?

Reply:
Let's pretend you wanna expose the graphql endpoint and the prometheus one on the same port:

```yaml
sandbox:
  listen: 0.0.0.0:4000
  path: /
  enabled: true # NOT MANDATORY?
telemetry:
  metrics:
    prometheus:
      listen: 127.0.0.1:4000
      path: /metrics
      enabled: true
```

With this setup /metrics would be exposed to WAN AFAICT, which would be a security issue.

`ensure_listenaddrs_consistency` makes sure the configuration doesn't have this, and raises an error if it's the case.

Reply:
Ok, so the way I imagined the feature was more aligned with this:

> The example you have in mind is currently not supported (this would require more efforts like binding to 0.0.0.0:4000 and then manually "route" the connections according to the configuration, so we decided to not support it yet.

But since it is not implemented yet we can end up with a security issue, right?

Reply:
We can’t, because the router explicitly refuses to start if that’s the case